File: planning-idsdp-pps-dpp-backend/src/main/java/com/ibcs/idsdp/dpptapp/services/TappLogFrameService.java
package com.ibcs.idsdp.dpptapp.services;

import com.ibcs.idsdp.common.web.dto.response.ResponseWithResults;
import com.ibcs.idsdp.dpptapp.web.dto.TappLogFrameDTO;

public interface TappLogFrameService {
     ResponseWithResults getTappLogFrame(String pcUuid);

     TappLogFrameDTO updateTappLogFrame(TappLogFrameDTO dto, String pcUuid);
}

File: planning-idsdp-pps-frontend/src/app/main/modules/dpp-tapp-management/services/tapp-log-frame.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {CrudRequestService} from '../../../core/services/crud-request.service';
import {environment} from '../../../../../environments/environment';
import {TappLogFrameModel} from '../models/tappLogFrame.model';
import {map} from "rxjs/operators";

@Injectable({
  providedIn: 'root'
})
export class TappLogFrameService extends CrudRequestService<TappLogFrameModel>{

    constructor(private http: HttpClient) {
        super(http, environment.ibcs.ppsDppBackendPoint + 'tapp-log-frame/');
    }

    getTappLogFrame(pcUuid){
        const url: string = environment.ibcs.ppsDppBackendPoint + 'tapp-log-frame/get-data' + '/' + pcUuid;
        return this.http.get(url).pipe(map((res: any) => {
            return res;
        }));
    }

    updateTappLogFrame(model, pcUuid){
        const url: string = environment.ibcs.ppsDppBackendPoint + 'tapp-log-frame/update-data' + '/' + pcUuid;
        return this.http.put(url, model).pipe(map((res: any) => {
            return res;
        }));
    }
}
